vyper/codegen/function_definitions/external_function.py

```diff
@@ -86,6 +86,9 @@ def handler_for(calldata_kwargs, default_kwargs):
         # a sequence of statements to strictify kwargs into memory
         ret = ["seq"]
 
+        # ensure calldata is at least of minimum length
+        ret.append(["assert", ["ge", "calldatasize", calldata_args_t.abi_type.min_size() + 4]])
```

Comment:
there need to be different checks depending if the data is static or has dynamic data.
```suggestion
        args_abi_t = calldata_args_t.abi_type
        calldata_min_size = args_abi_t.min_size() + 4
        if (args_abi_t.is_dynamic()):
            ret.append(["assert", ["ge", "calldatasize", calldata_min_size]])
        else:
            # stricter for static data
            ret.append(["assert", ["eq", "calldatasize", calldata_min_size]])
```